src/analysis/calculus/fderiv_measurable.lean

```diff
@@ -0,0 +1,455 @@
+  ∃ R > 0, ∀ r ∈ Ioo (0 : ℝ) R, x ∈ A f (fderiv 𝕜 f x) r ε :=
+begin
+  have := hx.has_fderiv_at,
+  simp only [has_fderiv_at, has_fderiv_at_filter, is_o_iff] at this,
+  rcases eventually_nhds_iff_ball.1 (this (half_pos hε)) with ⟨R, R_pos, hR⟩,
+  refine ⟨R, R_pos, λ r hr, _⟩,
+  have : r ∈ Ioc (r/2) r := ⟨half_lt_self hr.1, le_refl _⟩,
+  refine ⟨r, this, λ y z hy hz, _⟩,
+  calc  ∥f z - f y - (fderiv 𝕜 f x) (z - y)∥
+      = ∥(f z - f x - (fderiv 𝕜 f x) (z - x)) - (f y - f x - (fderiv 𝕜 f x) (y - x))∥ :
+    by { congr' 1, simp only [continuous_linear_map.map_sub], abel }
+  ... ≤ ∥(f z - f x - (fderiv 𝕜 f x) (z - x))∥ + ∥f y - f x - (fderiv 𝕜 f x) (y - x)∥ :
+    norm_sub_le _ _
+  ... ≤ ε / 2 * ∥z - x∥ + ε / 2 * ∥y - x∥ :
+    add_le_add (hR _ (lt_trans (mem_ball.1 hz) hr.2)) (hR _ (lt_trans (mem_ball.1 hy) hr.2))
+  ... ≤ ε / 2 * r + ε / 2 * r :
+    add_le_add
+      (mul_le_mul_of_nonneg_left (le_of_lt (mem_ball_iff_norm.1 hz)) (le_of_lt (half_pos hε)))
+      (mul_le_mul_of_nonneg_left (le_of_lt (mem_ball_iff_norm.1 hy)) (le_of_lt (half_pos hε)))
+  ... = ε * r : by ring
+end
+
+lemma norm_sub_le_of_mem_A {c : 𝕜} (hc : 1 < ∥c∥)
+  {r ε : ℝ} (hε : 0 < ε) (hr : 0 < r) {x : E} {L₁ L₂ : E →L[𝕜] F}
+  (h₁ : x ∈ A f L₁ r ε) (h₂ : x ∈ A f L₂ r ε) : ∥L₁ - L₂∥ ≤ 4 * ∥c∥ * ε :=
+begin
+  apply op_norm_le_of_shell (half_pos hr)
+    (mul_nonneg (mul_nonneg (by norm_num : (0 : ℝ) ≤ 4) (norm_nonneg _)) (le_of_lt hε)) hc,
+  assume y ley ylt,
+  rw [div_div_eq_div_mul,
+      div_le_iff' (mul_pos (by norm_num : (0 : ℝ) < 2) (zero_lt_one.trans hc))] at ley,
+  calc ∥(L₁ - L₂) y∥
+        = ∥(f (x + y) - f x - L₂ ((x + y) - x)) - (f (x + y) - f x - L₁ ((x + y) - x))∥ : by simp
+    ... ≤ ∥(f (x + y) - f x - L₂ ((x + y) - x))∥ + ∥(f (x + y) - f x - L₁ ((x + y) - x))∥ :
+      norm_sub_le _ _
+    ... ≤ ε * r + ε * r :
+      begin
+        apply add_le_add,
+        { apply le_of_mem_A h₂,
+          { simp only [le_of_lt (half_pos hr), mem_closed_ball, dist_self] },
+          { simp only [dist_eq_norm, add_sub_cancel', mem_closed_ball, ylt.le], } },
+        { apply le_of_mem_A h₁,
+          { simp only [le_of_lt (half_pos hr), mem_closed_ball, dist_self] },
+          { simp only [dist_eq_norm, add_sub_cancel', mem_closed_ball, ylt.le] } },
+      end
+    ... = 2 * ε * r : by ring
+    ... ≤ 2 * ε * (2 * ∥c∥ * ∥y∥) : mul_le_mul_of_nonneg_left ley (mul_nonneg (by norm_num) hε.le)
+    ... = 4 * ∥c∥ * ε * ∥y∥ : by ring
+end
+
+/-- Easy inclusion: a differentiability point with derivative in `K` belongs to `D f K`. -/
+lemma differentiable_set_subset_D : {x | differentiable_at 𝕜 f x ∧ fderiv 𝕜 f x ∈ K} ⊆ D f K :=
+begin
+  assume x hx,
+  rw [D, mem_Inter],
+  assume e,
+  have : (0 : ℝ) < (1/2) ^ e := pow_pos (by norm_num) _,
+  rcases mem_A_of_differentiable this hx.1 with ⟨R, R_pos, hR⟩,
+  obtain ⟨n, hn⟩ : ∃ (n : ℕ), (1/2) ^ n < R :=
+    exists_pow_lt_of_lt_one R_pos (by norm_num : (1 : ℝ)/2 < 1),
+  apply mem_Union.2 ⟨n, _⟩,
```

Comment:
I usually `simp only [mem_Union, mem_Inter]` right away in order to write just `use a` instead of `apply mem_Union.2 ⟨a, _⟩` later. But of course this is a matter of taste.

Reply:
Done